lib/bolt/plugin.rb

```diff
@@ -119,13 +119,9 @@ def boltdir
       end
     end
 
-    def self.setup(config, pal, pdb_client, analytics)
+    def self.setup(config, pal, analytics = Bolt::Analytics::NoopClient.new)
```

Comment:
If we set this default can we also remove passing the parameter when it's the noop client? I see it in `bolt_spec/bolt_context.rb`, `bolt/inventory.rb`, `bolt_spec/run.rb`, and a lot of tests. I know this PR is already big, but it's a nice simplification!